python/runtime/cudaq/algorithms/py_state.cpp

```diff
@@ -45,11 +49,39 @@ state pyGetState(kernel_builder<> &kernel, py::args args) {
 /// @brief Bind the get_state cudaq function
 void bindPyState(py::module &mod) {
 
-  py::class_<state>(mod, "State",
-                    "A representation of the internal simulation quantum state "
-                    "vector or density matrix.")
-      .def(py::init([](const py::buffer &b) {
-             py::buffer_info info = b.request();
+  py::class_<state>(
+      mod, "State", py::buffer_protocol(),
+      "A data-type representing the quantum state of the interal simulator. "
+      "Returns state vector by default. If qpu is set to `dm`, returns "
+      "density matrix.\n")
+      // Define a buffer model for this class to make it fully
+      // compatible with numpy.
+      .def_buffer([](state &self) -> py::buffer_info {
+        if (!self.is_density_matrix()) {
+          auto vector = self.get_data<Eigen::VectorXcd>();
```

Comment:
Instead of using Eigen here, you can just setup the state class to return the data itself. So keep the get_data method, but remove the template type, and have it just return `std::complex<double>*`

Reply:
You'll just need to add a way to get the number of elements for the vector and the number of rows and cols for the matrix